### PR TITLE
xeno defines redo to bypass bitfield limit

### PR DESCRIPTION
**Pull request by Mechanelize**
## About The Pull Request

redoing of xeno defines to bypass bitfield 21 line limit

## Why It's Good For The Game

allows for more xeno defines

## Changelog
:cl:
code: Split xeno related flags containing "CAN" into a new category called xeno_can_flags to bypass the 21 flag bitfield limit
/:cl:

### CODE REVIEW

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -485,28 +485,29 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
...
-#define CASTE_INNATE_PLASMA_REGEN (1<<13) // Xenos get full plasma regardless if they are on weeds or not
-#define CASTE_ACID_BLOOD (1<<14) //The acid blood effect which damages humans near xenos that take damage
-#define CASTE_CAN_HOLD_JELLY (1<<15)//whether we can hold fireproof jelly in our hands
-#define CASTE_IS_STRONG (1<<16)//can tear open acided walls without being big
-#define CASTE_CAN_CORRUPT_GENERATOR (1<<17) //Can corrupt a generator
-#define CASTE_IS_BUILDER (1<<18) //whether we are classified as a builder caste
-#define CASTE_CAN_BECOME_KING (1<<19) //Can be choose to become a king
-#define CASTE_CAN_RIDE_CRUSHER (1<<20) //Can ride a crusher
-#define CASTE_IS_A_MINION (1<<21) //That's a dumb ai
+#define CASTE_CAN_BE_LEADER (1<<49)
```
Comment (hyper2snyper) [inline code/__DEFINES/mobs.dm]: 49

File: code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm [hunk 2]
```diff
@@ -26,9 +26,9 @@
 	upgrade_threshold = TIER_THREE_YOUNG_THRESHOLD
 
 	// *** Flags *** //
-	caste_flags = CASTE_IS_INTELLIGENT|CASTE_CAN_HOLD_FACEHUGGERS|CASTE_CAN_HOLD_JELLY|CASTE_CAN_CORRUPT_GENERATOR|CASTE_IS_BUILDER|CASTE_CAN_BE_GIVEN_PLASMA
-
+	caste_flags = CASTE_IS_INTELLIGENT|CASTE_IS_BUILDER
 	can_hold_eggs = CAN_HOLD_TWO_HANDS
+	can_flags = |CASTE_CAN_HOLD_FACEHUGGERS|CASTE_CAN_HOLD_JELLY|CASTE_CAN_CORRUPT_GENERATOR|CASTE_CAN_BE_GIVEN_PLASMA
```
Comment (hyper2snyper) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/castedatum_queen.dm:31]:
```suggestion
	can_flags = CASTE_CAN_HOLD_FACEHUGGERS|CASTE_CAN_HOLD_JELLY|CASTE_CAN_CORRUPT_GENERATOR|CASTE_CAN_BE_GIVEN_PLASMA
```

**Review by hyper2snyper — CHANGES_REQUESTED** [review]
I also didnt see a definition for can_flags anywhere.
btw 'can_flags' is a bad name
make it something like 'xeno_ability_flags'

**Comment by KornFlaks** [comment]
> make it something like 'xeno_ability_flags'

Definitely not. Since we have xeno action flags (XACT_USE) and abilities are filed under that.

Maybe xeno_mod_flags for modifiers. Honestly, xeno_can_flags is the most appropriate.

**Comment by Al-1ce** [comment]
I have the strangest feeling that adjusting xeno flags will result in a shitstorm of bugs.

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Atomize the balance change please

**Pull request merged**